sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -139,51 +131,39 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
     sessionOptions: CreateSessionReceiverOptions,
     retryOptions: RetryOptions = {}
   ): Promise<SessionReceiver<ReceivedMessageT>> {
+    context.isSessionEnabled = true;
+    if (sessionOptions.sessionId != undefined) {
+      sessionOptions.sessionId = String(sessionOptions.sessionId);
+    }
+    const messageSession = await MessageSession.create(context, {
+      sessionId: sessionOptions.sessionId,
+      autoRenewLockDurationInMs: sessionOptions.autoRenewLockDurationInMs,
+      receiveMode: convertToInternalReceiveMode(receiveMode)
+    });
     const sessionReceiver = new SessionReceiverImpl<ReceivedMessageT>(
+      messageSession,
       context,
       receiveMode,
-      sessionOptions,
       retryOptions
     );
-
-    await sessionReceiver._createMessageSessionIfDoesntExist();
     return sessionReceiver;
   }
 
   private _throwIfReceiverOrConnectionClosed(): void {
     throwErrorIfConnectionClosed(this._context.namespace);
-    if (this.isClosed) {
-      const errorMessage = getReceiverClosedErrorMsg(
-        this._context.entityPath,
-        this._context.isClosed,
-        this.sessionId!
-      );
+    if (this._isClosed) {
+      const errorMessage = getReceiverClosedErrorMsg(this._context.entityPath, this.sessionId!);
       const error = new Error(errorMessage);
       log.error(`[${this._context.namespace.connectionId}] %O`, error);
       throw error;
     }
-  }
-
-  private async _createMessageSessionIfDoesntExist(): Promise<void> {
-    // TODO - pass timeout for MessageSession creation
-    if (this._messageSession) {
-      return;
-    }
-    this._context.isSessionEnabled = true;
-    this._messageSession = await MessageSession.create(this._context, {
-      sessionId: this._sessionOptions.sessionId,
-      autoRenewLockDurationInMs: this._sessionOptions.autoRenewLockDurationInMs,
-      receiveMode: convertToInternalReceiveMode(this.receiveMode)
-    });
-    // By this point, we should have a valid sessionId on the messageSession
-    // If not, the receiver cannot be used, so throw error.
-    if (this._messageSession.sessionId == null) {
-      const error = new Error("Something went wrong. Cannot lock a session.");
-      log.error(`[${this._context.namespace.connectionId}] %O`, error);
-      throw error;
+    if (!this._context.messageSessions[this.sessionId] || !this._messageSession.isOpen()) {
```

Comment:
This check is already made in `isClosed`. Can we refactor as below?
```js
if (this.isClosed) {
    if (this._isClosed) { 
       // call getRecieverClosedErrorMsg() and throw it 
    }
    // throw SessionLockLostError here

}

Reply:
Can do!

Reply:
Done